spec/unit/info_service_spec.rb

```diff
@@ -57,11 +57,27 @@
         end
       end
 
-      context 'For a module with multiple implemenations' do
+      context 'For a module with multiple implemenations and files' do
+        let(:other_mod_name) { "shell_helpers" }
         let(:metadata) {
           { "implementations" => [
-            {"name" => "thingtask.rb", "requirements" => ["puppet_agent"] },
-            {"name" => "thingtask.sh", "requirements" => ["shell"] } ] } }
+            {"name" => "thingtask.rb", "requirements" => ["puppet_agent"],
+             "files" => ["#{mod_name}/lib/puppet/providers/prov.rb"]},
```

Comment:
Let's drop `/prov.rb` from this specification, which will give us a test with a directory containing files.